app/views/components/_slider.html.erb

```diff
@@ -6,9 +6,13 @@
   step = local_assigns[:step] || 10
   initial_value = local_assigns[:initial_value] || 0
   option = local_assigns[:option] ? "slider--#{local_assigns[:option]}" : ""
+  target = local_assigns[:target] ? "data-target=#{local_assigns[:target]}" : ""
+  text_before = local_assigns[:text_before] || ""
+  text_after = local_assigns[:text_after] || ""
 %>
 
-<div class="slider <%= option %>" data-slider id="<%= id %>">
+<div class="slider <%= option %>" data-slider id="<%= id %>" <%= target %>>
+  <%= text_before %>
```

Comment:
Rails supports [layouts for partials](http://guides.rubyonrails.org/layouts_and_rendering.html#partial-layouts). They might work for you here?